stdlib/listLabels.mli

```diff
@@ -104,8 +104,6 @@ val init : len:int -> f:(int -> 'a) -> 'a list
 
 val append : 'a list -> 'a list -> 'a list
 (** Concatenate two lists. Same function as the infix operator [@].
```

Comment:
While we are here could we improve this doc string to tell the append order and link to the operator with `{!Stdlib.(@)}`: 
```
[append l0 l1] appends [l1] to [l0]. Same function as the infix operators {!Stdlib.(@)}.
```

Reply:
Done; thanks!